airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/manifest_component_transformer.py

```diff
@@ -103,7 +105,7 @@ def propagate_types_and_parameters(
                 propagated_component["type"] = found_type
 
         # When there is no resolved type, we're not processing a component (likely a regular object) and don't need to propagate parameters
-        if "type" not in propagated_component:
+        if "type" not in propagated_component or propagated_component["type"] in _PROPAGATION_EXCLUSION_TYPES:
```

Comment:
A minor thought I had here is that we actually don't want to propagate things is the `schema` field of the `InlineSchemaLoader` as opposed to the whole schema loader.

And that is because the `schema` field is not actually a component at all. The same for ohter compoents. For a field something like JSON body field we wouldn't want to accidentally inject all the params there.

I did some snooping and I think this original check is actually not doing what we want. Because even for the `InlineSchemaLoader.schema` field, it has a `type` field set to `object`. So this condition never triggers. I haven't thoroughly vetted this, just re-running tests, but changing the condition to: `if "type" not in propagated_component or propagated_component.get("type") == "object":`. 

So then the `InlineSchemaLoader` can still receive parameters, but it doesn't pass them to `schema. If this is indeed accurate, then I prefer this solution because it means less one-of logic

Reply:
This makes sense to me. Let me dig down and see if there are edge cases for that solution

Reply:
It feel like it makes sense. I added a comment to clarify the scope of this check